src/tcgwars/logic/impl/gen3/Deoxys.groovy

```diff
@@ -2880,13 +2880,10 @@ public enum Deoxys implements LogicCardInfo {
             energyCost C, C
             onAttack {
               def pcs = defending
-              if(opp.bench){
-                if(confirm("Switch 1 of your opponent’s Benched Pokémon with the Defending Pokémon before doing damage?")){
-                  pcs = opp.bench.select()
-                  sw defending, pcs
-                }
+              if(opp.bench && confirm("Switch 1 of your opponent's Benched Pokémon with the Defending Pokémon?")){
+                def target = opp.bench.select("Select the new Active Pokémon.")
+                if ( sw2(target) ) { pcs = target }
               }
-
               damage 20, pcs
```

Comment:
actually this assignment logic is not necessary `damage 20` will always register the damage to the current active (defending) pokemon

Reply:
That sounds good; didn't even think about it after the initial tests with roselia (since first bug was it applying to the unswitched active). Will correct this in a bit.

Reply:
(Never mind that, mixed this with another card when reading the email; still fixing it tho, just different reason :P)